components/Deploy.tsx

```diff
@@ -0,0 +1,262 @@
+import { FC, useState, useEffect } from 'react'
+import WebMonetizationLoader from './WebMonetizationLoader'
+import { parse } from '@prantlf/jsonlint'
+import { dump, safeLoad } from 'js-yaml'
+import AceEditor, {IAnnotation} from 'react-ace'
+
+import 'ace-builds/src-noconflict/mode-json'
+import 'ace-builds/src-noconflict/mode-yaml'
+import 'ace-builds/src-noconflict/theme-github'
+
+const { useMonetizationState } = require('react-web-monetization')
+
+interface DeployProps {
+  codiusHostURI: string
+}
+
+const Deploy: FC<DeployProps> = (props: DeployProps) => {
+  const defaultService = {
+    spec: {
+      containers: [
+        {
+          name: "app",
+          image: "nginx@sha256:3e2ffcf0edca2a4e9b24ca442d227baea7b7f0e33ad654ef1eb806fbd9bedcf0",
+          command: [
+            "nginx",
+            "-g",
+            "daemon off;"
+          ],
+          env: [
+            {
+              name: "PUBLIC_VARIABLE",
+              value: "hello world"
+            },
+            {
+              name: "PRIVATE_VARIABLE",
+              valueFrom: {
+                secretKeyRef: {
+                  key: "password"
+                }
+              }
+            }
+          ],
+          readinessProbe: {
+            httpGet: {
+              path: "/",
+              port: 80
+            },
+            failureThreshold: 10,
+          }
+        }
+      ],
+      port: 80
+    },
+    secretData: {
+      nonce: "123456789abcdef",
+      password: "super secret"
+    }
+  }
+
+  const hostUrl = new URL(props.codiusHostURI)
+  const host = hostUrl.host
+  const protocol = hostUrl.protocol
+
+  const { requestId } = useMonetizationState()
+
+  const [name, setName] = useState('')
```

Comment:
this is a really big component with a lot of pieces of state. If we plan on maintaining this frontend in the future then it would be a good idea to break it into smaller components and use a central store for the state. You could look into using redux or recoil (less boilerplate), or even react contexts. Then you can have all these different inputs modify your state and trigger async actions without having to put them all in the same component.